src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -8360,6 +8360,363 @@ .maxstack  2
+  IL_0017:  ldarg.0
+  IL_0018:  call       ""object..ctor()""
+  IL_001d:  nop
+  IL_001e:  ret
+}
+");
+        }
+
+        [Fact]
+        public void ExpressionVariable_InQuery()
+        {
+            var source0 = MarkedSource(@"
+using System.Linq;
+public class Program
+{
+    static int M(int x, out int y) { y = 42; return 43; }
+    static void N()
+    {
+        var query =
+            from a in new int[] { 1, 2 }
+            select M(a, out int <N:0>x</N:0>) + x + M(a, out int <N:1>y</N:1>);
+    }
+}");
+            var source1 = MarkedSource(@"
+using System.Linq;
+public class Program
+{
+    static int M(int x, out int y) { y = 42; return 43; }
+    static void N()
+    {
+        var query =
+            from a in new int[] { 1, 2 }
+            select M(a, out int <N:0>x</N:0>) + x;
+    }
+}");
+             var compilation0 = CreateCompilation(source0.Tree, options: ComSafeDebugDll);
+            var compilation1 = compilation0.WithSource(source1.Tree);
+            var compilation2 = compilation1.WithSource(source0.Tree);
+
+            var n0 = compilation0.GetMember<MethodSymbol>("Program.N");
+            var n1 = compilation1.GetMember<MethodSymbol>("Program.N");
+            var n2 = compilation2.GetMember<MethodSymbol>("Program.N");
+
+            var v0 = CompileAndVerify(compilation0);
+            v0.VerifyIL("Program.N()", @"
+{
+  // Code size       53 (0x35)
+  .maxstack  4
+  .locals init (System.Collections.Generic.IEnumerable<int> V_0) //query
+  IL_0000:  nop
+  IL_0001:  ldc.i4.2
+  IL_0002:  newarr     ""int""
+  IL_0007:  dup
+  IL_0008:  ldc.i4.0
+  IL_0009:  ldc.i4.1
+  IL_000a:  stelem.i4
+  IL_000b:  dup
+  IL_000c:  ldc.i4.1
+  IL_000d:  ldc.i4.2
+  IL_000e:  stelem.i4
+  IL_000f:  ldsfld     ""System.Func<int, int> Program.<>c.<>9__1_0""
+  IL_0014:  dup
+  IL_0015:  brtrue.s   IL_002e
+  IL_0017:  pop
+  IL_0018:  ldsfld     ""Program.<>c Program.<>c.<>9""
+  IL_001d:  ldftn      ""int Program.<>c.<N>b__1_0(int)""
+  IL_0023:  newobj     ""System.Func<int, int>..ctor(object, System.IntPtr)""
+  IL_0028:  dup
+  IL_0029:  stsfld     ""System.Func<int, int> Program.<>c.<>9__1_0""
+  IL_002e:  call       ""System.Collections.Generic.IEnumerable<int> System.Linq.Enumerable.Select<int, int>(System.Collections.Generic.IEnumerable<int>, System.Func<int, int>)""
+  IL_0033:  stloc.0
+  IL_0034:  ret
+}
+");
+
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, n0, n1, GetSyntaxMapFromMarkers(source0, source1), preserveLocalVariables: true)));
+
+            diff1.VerifyIL("Program.N()", @"
+{
+  // Code size       53 (0x35)
+  .maxstack  4
+  .locals init ([unchanged] V_0,
```

Comment:
>[unchanged] V_0, [](start = 16, length = 16)

This indicates that the query variable was not reused. This is likely just because you're missing the markers <N:#> around the expression in `var query = ...` declaration.
Also, as Aleksey points out we should test how the lambda was updated. You'll need more markers to map the clauses of the query.
 See [http://source.roslyn.io/#Roslyn.Compilers.CSharp.Emit.UnitTests/Emit/EditAndContinue/EditAndContinueTests.cs,5338](http://source.roslyn.io/#Roslyn.Compilers.CSharp.Emit.UnitTests/Emit/EditAndContinue/EditAndContinueTests.cs,5338) for another LINQ EnC test.